__tests__/topics.test.js

```diff
@@ -0,0 +1,46 @@
+const request = require("supertest");
+
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data/index");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("GET /api/topics", () => {
+  test("responds with array of topic objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(topics).toHaveLength(3);
+        topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug", expect.any(String));
+          expect(topic).toHaveProperty("description", expect.any(String));
+        });
+      });
+  });
+  test("responds with error if no topic objects in database", () => {
+    return db
+      .query(
+        `DELETE FROM comments;
+         DELETE FROM articles;
+         DELETE FROM topics;`
+      )
+      .then(() => {
+        return request(app)
+          .get("/api/topics")
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Not found");
+          });
```

Comment:
I would remove this error test. It's not necessarily an error if there are no records in the database, there just aren't any records and that's fine :) Errors are usually testing for improper use of the server. 

For example If I asked to see all the topics and there weren't any, I'd be more than happy to receive an empty array :)

Reply:
You might want to consider a separate 404 - not found test (not for this endpoint). You could make some request to non existent URLs "/api/notanendpoint" and simply expect back a 404 status code, express sends this for you, but it's important to test for it :)